clang/include/clang/Basic/Attr.td

```diff
@@ -4061,3 +4061,20 @@ def NoRandomizeLayout : InheritableAttr {
   let LangOpts = [COnly];
 }
 def : MutualExclusions<[RandomizeLayout, NoRandomizeLayout]>;
+
+def FunctionReturnThunks : InheritableAttr,
+    TargetSpecificAttr<TargetAnyX86> {
+  let Spellings = [GCC<"function_return">];
+  let Args = [EnumArgument<"ThunkType", "Kind",
+    ["keep", "thunk-extern"],
+    ["Keep", "Extern"]
+  >];
+  let Subjects = SubjectList<[Function]>;
+  let Documentation = [FunctionReturnThunksDocs];
+}
```

Comment:
This was not part of https://reviews.llvm.org/D135851. It was introduced in https://reviews.llvm.org/D129572 which also modified `pragma-attribute-supported-attributes-list.test`. Without those changes, the test is broken. I would recommend recoming this `def FunctionReturnThunks`, since most of its implementation is not available.